internal/db/db_test.go

```diff
@@ -40,115 +44,163 @@ func (s *dbSuite) Test_waitUpgrade() {
 	}{
 		{
 			name:              "No upgrade, no other nodes",
-			upgradedLocalInfo: versions{schemaInt: 0, schemaExt: 0},
+			upgradedLocalInfo: versions{schemaInt: 0, schemaExt: 0, apiExtensionInt: "internal:ext_a,internal:ext_b", apiExtensionExt: "ext_c,ext_d"},
```

Comment:
I think having all these API updates in this test is a bit overkill. You've added (I think) the same extensions to each entry, but I think we should limit the test input to only the values that we want to change across the tests, otherwise it just makes it hard to read what the test is doing. If you need to set a default value, you can do that in the actual test loop below.

I think maybe you can keep this test the same and rename it to `waitUpgradeSchema`, and then add a new test case for `waitUpgradeAPI` which mainly changes the `apiExtensionsInt` and `apiExtensionsExt` in all the same cases as this test. That might make it easier to follow what each test is doing as there's not so many moving variables.

We can also have a 3rd test function that checks if both API and Schema versions are being updated at the same time.

Reply:
fixed.